src/App.tsx

```diff
@@ -17,18 +40,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                titleFilter={titleFilter}
+                handleTitleFilterChange={setTitleFilter}
+                statusFilter={statusFilter}
+                handleStatusFilterChange={setStatusFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {!isLoading && (
+                <TodoList
+                  todos={displayTodos}
+                  handleSelectTodo={setSelectedTodo}
+                  selectedTodo={selectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {
+        selectedTodo && (
+          <TodoModal
+            selectedTodo={selectedTodo}
+            handleClose={() => setSelectedTodo(null)}
```

Comment:
Nitpick: pass setter without `null` argument here and use `null` in the child component. This way a person who reads the code of `TodoModal` will understand what we actually set there.